### PR TITLE
[google_maps_flutter] Adds support for holes in polygon overlays to the Google Maps plugin

### PR DESCRIPTION
**Pull request by sanekyy**
## Description

Adds support for holes in polygon to the Google Maps plugin

This is trivial. I just added the "hole" property to an already existing polygon.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by otopba** [comment]
@iskakaushik 
Hello! 
Could you check it please? This PR is very necessary for my project. 
Thank you 🙏

File: packages/google_maps_flutter/pubspec.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: google_maps_flutter
 description: A Flutter plugin for integrating Google Maps in iOS and Android applications.
 author: Flutter Team <flutter-dev@googlegroups.com>
 homepage: https://github.com/flutter/plugins/tree/master/packages/google_maps_flutter
-version: 0.5.19+2
```
Comment (otopba) [inline packages/google_maps_flutter/pubspec.yaml]: This is not good to change version inside PR

File: packages/google_maps_flutter/lib/src/polygon.dart [hunk 2]
```diff
@@ -67,6 +68,11 @@ class Polygon {
   /// default; to form a closed polygon, the start and end points must be the same.
   final List<LatLng> points;
 
+  /// The vertices of the holes to be cut out of polygon.
+  ///
+  /// Line segments of each points of hole are drawn inside polygon between consecutive hole points.
```
Comment (iskakaushik) [inline packages/google_maps_flutter/lib/src/polygon.dart]: Additional documentation as to what a hole is would be helpful. Also something about the sequence in which the lat lngs are supposed to be input. Something along the lines of:

To create an empty area within a polygon, you need to create two paths, one inside the other. To create the hole, the coordinates defining the inner path must be in the opposite order to those defining the outer path. For example, if the coordinates of the outer path are in clockwise order then the inner path must be counter-clockwise.

I copied it over from https://developers.google.com/maps/documentation/javascript/shapes#polygon_hole

File: packages/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/PolygonController.java [hunk 3]
```diff
@@ -46,6 +46,10 @@ public void setPoints(List<LatLng> points) {
     polygon.setPoints(points);
   }
 
+  public void setHoles(List<List<LatLng>> holes) {
```
Comment (iskakaushik) [inline packages/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/PolygonController.java]: what happens if the holes lie outside the polygon? does the google maps api throw some error?

**Comment by collinjackson** [comment]
@sanekyy Thanks for the contribution. If you'd like us to land this, could you please address the reviewer feedback when you get a chance?

File: packages/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/PolygonController.java (hunk 3 above)
Comment (sanekyy) [reply packages/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/PolygonController.java]: Documentation from https://developers.google.com/maps/documentation/android-sdk/shapes:

If the hole intersects the outline of the polygon, the polygon will be rendered without any fill.

File: packages/google_maps_flutter/lib/src/polygon.dart (hunk 2 above)
Comment (sanekyy) [reply packages/google_maps_flutter/lib/src/polygon.dart]: done

**Comment by sanekyy** [comment]
Thanks for review. I'm ready to write simple test. But before start writing I tried to run existing tests and tests failed.

I added google map API_KEY and run ```flutter drive test_driver/google_maps.dart``` from ```packages/google_maps_flutter/example```

Can you say is it some my mistake or this tests not working?

Here is an output:

```
➜  example git:(polygon_holes) ✗ flutter drive test_driver/google_maps.dart                                                                     
Using device Redmi 4X.
Starting application: test_driver/google_maps.dart
Initializing gradle...                                              0.6s
Resolving dependencies...                                           1.0s
Installing build/app/outputs/apk/app.apk...                             
  149.7s (!)                                       
Running Gradle task 'assembleDebug'...                                  
Running Gradle task 'assembleDebug'... Done                        13.1s
Built build/app/outputs/apk/debug/app-debug.apk.
Installing build/app/outputs/apk/app.apk...                        16.7s
I/flutter (28257): Observatory listening on http://127.0.0.1:52908/HzLMoyYt-h0=/
[info ] FlutterDriver: Connecting to Flutter application at http://127.0.0.1:60948/HzLMoyYt-h0=/
[trace] FlutterDriver: Isolate found with number: 1317353945037787
[trace] FlutterDriver: Isolate is paused at start.
[trace] FlutterDriver: Attempting to resume isolate
[trace] FlutterDriver: Waiting for service extension
I/flutter (28257): 00:00 +0: testCompassToggle
[info ] FlutterDriver: Connected to Flutter application.
I/flutter (28257): 00:02 +1: testMapToolbarToggle
I/flutter (28257): 00:02 +2: updateMinMaxZoomLevels
I/flutter (28257): 00:03 +2: updateMinMaxZoomLevels [E]
I/flutter (28257):   Expected: MinMaxZoomPreference:<MinMaxZoomPreference(minZoom: 2.0, maxZoom: 4.0)>
I/flutter (28257):     Actual: MinMaxZoomPreference:<MinMaxZoomPreference(minZoom: 0.0, maxZoom: 21.0)>
I/flutter (28257):   
I/flutter (28257):   package:test_api/src/frontend/expect.dart 152:30                                                                fail
I/flutter (28257):   package:test_api/src/frontend/expect.dart 146:3                                                                 _expect
I/flutter (28257):   package:test_api/src/frontend/expect.dart 59:3                                                                  expect
I/flutter (28257):   package:flutter_test/src/widget_tester.dart 229:3                                                               expect
I/flutter (28257):   Users/sanekyy/Projects/flutter/plugins/packages/google_maps_flutter/example/test_driver/google_maps.dart 131:5  main.<fn>
I/flutter (28257):   ===== asynchronous gap ===========================
I/flutter (28257):   dart:async/zone.dart 1062:19                                                                                    _CustomZone.registerBinaryCallback
I/flutter (28257):   dart:async-patch/async_patch.dart 80:23                                                                         _asyncErrorWrapperHelper
I/flutter (28257):   package:test_api/src/backend/invoker.dart                                
I/flutter (28257): 00:03 +2 -1: testZoomGesturesEnabled
I/flutter (28257): 00:04 +3 -1: testRotateGesturesEnabled
I/flutter (28257): 00:04 +4 -1: testTiltGesturesEnabled
I/flutter (28257): 00:05 +5 -1: testScrollGesturesEnabled
I/flutter (28257): 00:05 +6 -1: testGetVisibleRegion
I/flutter (28257): 00:08 +7 -1: testTraffic
I/flutter (28257): 00:09 +8 -1: testMyLocationButtonToggle
I/flutter (28257): 00:09 +8 -1: testMyLocationButtonToggle [E]
I/flutter (28257):   Expected: <false>
I/flutter (28257):     Actual: <true>
I/flutter (28257):   
I/flutter (28257):   package:test_api/src/frontend/expect.dart 152:30                                                                fail
I/flutter (28257):   package:test_api/src/frontend/expect.dart 146:3                                                                 _expect
I/flutter (28257):   package:test_api/src/frontend/expect.dart 59:3                                                                  expect
I/flutter (28257):   package:flutter_test/src/widget_tester.dart 229:3                                                               expect
I/flutter (28257):   Users/sanekyy/Projects/flutter/plugins/packages/google_maps_flutter/example/test_driver/google_maps.dart 442:5  main.<fn>
I/flutter (28257):   ===== asynchronous gap ===========================
I/flutter (28257):   dart:async/zone.dart 1062:19                                                                                    _CustomZone.registerBinaryCallback
I/flutter (28257):   dart:async-patch/async_patch.dart 80:23                                                                         _asyncErrorWrapperHelper
I/flutter (28257):   package:test_api/src/backend/invoker.dart                                
I/flutter (28257): 00:10 +8 -2: testMyLocationButton initial value false
I/flutter (28257): 00:10 +8 -2: testMyLocationButton initial value false [E]
I/flutter (28257):   Expected: <false>
I/flutter (28257):     Actual: <true>
I/flutter (28257):   
I/flutter (28257):   package:test_api/src/frontend/expect.dart 152:30                                                                fail
I/flutter (28257):   package:test_api/src/frontend/expect.dart 146:3                                                                 _expect
I/flutter (28257):   package:test_api/src/frontend/expect.dart 59:3                                                                  expect
I/flutter (28257):   package:flutter_test/src/widget_tester.dart 229:3                                                               expect
I/flutter (28257):   Users/sanekyy/Projects/flutter/plugins/packages/google_maps_flutter/example/test_driver/google_maps.dart 469:5  main.<fn>
I/flutter (28257):   ===== asynchronous gap ===========================
I/flutter (28257):   dart:async/zone.dart 1062:19                                                                                    _CustomZone.registerBinaryCallback
I/flutter (28257):   dart:async-patch/async_patch.dart 80:23                                                                         _asyncErrorWrapperHelper
I/flutter (28257):   package:test_api/src/backend/invoker.dart                                
I/flutter (28257): 00:10 +8 -3: testMyLocationButton initial value true
I/flutter (28257): 00:11 +9 -3: testSetMapStyle valid Json String
I/flutter (28257): 00:11 +10 -3: testSetMapStyle invalid Json String
I/flutter (28257): 00:11 +10 -3: testSetMapStyle invalid Json String [E]
I/flutter (28257):   Expected: 'The data couldn’t be read because it isn’t in the correct format.'
I/flutter (28257):     Actual: 'Unable to set the map style. Please check console logs for errors.'
I/flutter (28257):      Which: is different.
I/flutter (28257):             Expected: The data c ...
I/flutter (28257):               Actual: Unable to  ...
I/flutter (28257):                       ^
I/flutter (28257):              Differ at offset 0
I/flutter (28257):   
I/flutter (28257):   package:test_api/src/frontend/expect.dart 152:30                                                                fail
I/flutter (28257):   package:test_api/src/frontend/expect.dart 146:3                                                                 _expect
I/flutter (28257):   package:test_api/src/frontend/expect.dart 59:3                                                                  expect
I/flutter (28257):   package:flutter_test/src/widget_tester.dart 229:3                                                               expect
I/flutter (28257):   Users/sanekyy/Projects/flutter/plugins/packages/google_maps_flutter/example/test_driver/google_maps.dart 543:7  main.<fn>
I/flutter (28257):   ===== asynchronous gap ===========================
I/flutter (28257):   dart:async/zone.dart 1062:19                                                                                    _CustomZone.registerBinaryCallback
I/flutter (28257):   dart:async-patch/async_patch.dart 80:23                                                                         _asyncErrorWrapperHelper
I/flutter (28257):   package:test_api/src/backend/invoker.dart                                
I/flutter (28257): 00:11 +10 -4: testSetMapStyle null string
I/flutter (28257): 00:12 +11 -4: (tearDownAll)
I/flutter (28257): 00:12 +12 -4: Some tests failed.
Stopping application instance.

```

**Comment by jl-gogovapps** [comment]
This looks like great work, curious for a status update to get it over the finish line?  Thanks!

**Comment by sanekyy** [comment]
@jl-gogovapps, yes of course

**Comment by sanekyy** [comment]
`publishable` task failed with:


```
Checking that google_maps_flutter can be published.
Suggestions:
* line 15, column 1 of lib/google_maps_flutter.dart: This package does not have collection in the `dependencies` section of `pubspec.yaml`.
     ╷
  15 │ import 'package:collection/collection.dart';
     │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     ╵

Package has 1 warning.
```
I added this import to use `DeepCollectionEquality` class from `collection`.

To fix this problem I added `collection: ^1.14.11` in dependencies section in yaml. I use this version because flutter_driver from sdk already depends on collection 1.14.11.

Is it a right way?

**Comment by sanekyy** [comment]
@jl-gogovapps 

Hello. Can you check my PR?

**Comment by jl-gogovapps** [comment]
Sure, I'll have some time today - however not sure I have any clout in this community :)

**Comment by jl-gogovapps** [comment]
@sanekyy this all seems fine, just want to run something by you

I'm unable to get "shading" to work correctly for complex polygons with holes , unless I use Google Maps Beta 3.0.0.  I think this is a bug elsewhere, not on this branch.  My drawing code handles MultiPolygons, and is essentially:

```
    Set<Polygon> grayout = <Polygon>{};

    double mercatorProjection = 85.05113;
    double mercatorProjectionSouth = -85.05113;
    double antiMeridian = 180;
    double antiMeridianSouth = -180;
    double primeMeridian = 0;
      
    List<List<LatLng>> positives = new List<List<LatLng>>();

    multipolygonCoordinates().forEach((outerPaths) {
      List<LatLng> outerPath = arrayToPath(outerPaths[0]);
      positives.add(outerPath);

      outerPaths.skip(1).forEach((innerPath) {
        positives.add(arrayToPath(innerPath));
      });
    });

    grayout.add(new Polygon(
            polygonId: PolygonId('world1'),
            strokeColor: Colors.black,
            strokeWidth: toStrokeWidth(2),
            fillColor: Color(7829368).withOpacity(0.4),
            holes: positives,
            points: [
              new LatLng(mercatorProjection, antiMeridianSouth),
              new LatLng(mercatorProjectionSouth, antiMeridianSouth),
              new LatLng(mercatorProjectionSouth, primeMeridian),
              new LatLng(mercatorProjection, primeMeridian),
              new LatLng(mercatorProjection, antiMeridianSouth)
            ]
    ));
    return grayout;
```

side note - i'm aware this is only shading the western hemisphere

**Comment by sanekyy** [comment]
Hello @iskakaushik 
Can you help me with merging this PR please?

**Comment by lbaey** [comment]
Hoping for this to land quickly, this is key feature in our project. 🙂 Thx.

File: packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,7 @@
+## 0.5.24
+
+* Add support for holes in Polygons
```
Comment (cyanglaz) [inline packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md]: nits:
```suggestion
* Add support for holes in Polygons.
```

File: packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/Convert.java [hunk 5]
```diff
@@ -559,6 +563,16 @@ static String interpretCircleOptions(Object o, CircleOptionsSink sink) {
     return points;
   }
 
+  private static List<List<LatLng>> toHoles(Object o) {
+    final List<?> data = toList(o);
+    final List<List<LatLng>> holes = new ArrayList<>(data.size());
+
+    for (Object ob : data) {
```
Comment (cyanglaz) [inline packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/Convert.java]: maybe make the code more self explanatory by renaming the `ob` to something more meaningful?

File: packages/google_maps_flutter/google_maps_flutter/example/lib/place_polygon.dart [hunk 6]
```diff
@@ -96,6 +96,8 @@ class PlacePolygonBodyState extends State<PlacePolygonBody> {
 
     setState(() {
       polygons[polygonId] = polygon;
+      polygonOffsets[polygonId] = _polygonIdCounter.ceilToDouble();
+      _polygonIdCounter++;
```
Comment (cyanglaz) [inline packages/google_maps_flutter/google_maps_flutter/example/lib/place_polygon.dart]: Can we add comments on what does this code do?

File: packages/google_maps_flutter/google_maps_flutter/lib/src/polygon.dart [hunk 7]
```diff
@@ -179,6 +195,18 @@ class Polygon {
     }
     return result;
   }
+
+  dynamic _holesToJson() {
```
Comment (cyanglaz) [inline packages/google_maps_flutter/google_maps_flutter/lib/src/polygon.dart]: We should be able to just return `List<dynamic>`

File: packages/google_maps_flutter/google_maps_flutter/example/lib/place_polygon.dart (hunk 6 above)
Comment (sanekyy) [reply packages/google_maps_flutter/google_maps_flutter/example/lib/place_polygon.dart]: done

File: packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/Convert.java (hunk 5 above)
Comment (sanekyy) [reply packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/Convert.java]: Done.

This style of naming is present in all methods in this class. So maybe it worth to refactor in separate PR.

File: packages/google_maps_flutter/google_maps_flutter/lib/src/polygon.dart (hunk 7 above)
Comment (sanekyy) [reply packages/google_maps_flutter/google_maps_flutter/lib/src/polygon.dart]: done

**Comment by sanekyy** [comment]
Hello @cyanglaz. 

Thanks for review.
Fixed all suggestions and remarks.

**Comment by sanekyy** [comment]
Hey @cyanglaz. Tests added. Ready to merge.

**Comment by JayPerfetto** [comment]
When's this getting merged? I also need it

**Comment by duzenko** [comment]
https://previews.123rf.com/images/vectordivider/vectordivider1612/vectordivider161200094/68411890-one-year-anniversary-celebration-golden-vintage-logotype-first-anniversary-gold-label-in-floral-wrea.jpg

**Comment by jl-gogovapps** [comment]
@sanekyy I saw you were also looking at some performance related PRs (https://github.com/flutter/plugins/pull/2479), and the platform initialization code has been refactored.  I have a few questions:
- Does that negatively impact this PR, in terms of mergability?  
- I see there is still a "needs test harness" label, is that holding anything up?
- Have you had to, or been able to, combine these changes with performance improvements in the other PRs?  Curious what the results might be, as performance issues regarding the map initialization are pretty significant in our app.

Again, thanks for the great work on this, it has been massive for us.  I hope it can get merged soon!

**Comment by mattcoker** [comment]
Agreed, this is a mission-critical piece of functionality for myself as well. It looks like the only thing blocking now is the conflicting files and the reliance on `google_maps_flutter_platform_interface` through @sanekyy's fork of `flutter/plugins`.

Is there anything else that needs to happen to get some eyes on this and get this merged?

**Comment by taeilyoon** [comment]
how can i use it?

**Comment by sanekyy** [comment]
> how can i use it?

Hi @taeilyoon. You can use my fork.

**Comment by jl-gogovapps** [comment]
@sanekyy might you have a chance to rebase this on the latest upstream master? thanks regardless!

**Comment by taeilyoon** [comment]
> > how can i use it?
> 
> Hi @taeilyoon. You can use my fork.

Thank you for you help
doesn’t it  run on ios?
or
can you give me sample on ios ?

it run android perfect

**Comment by fnicastri** [comment]
This is a very important feature

**Comment by cyanglaz** [comment]
@sanekyy Sorry looks like this PR was ready to merge and it fell off my radar. Do you mind rebase the PR and ping me? I'll merge it.

**Comment by sanekyy** [comment]
Hi @cyanglaz, okay, I'll do it.

**Comment by sanekyy** [comment]
@cyanglaz Ping. Done.

**Comment by jl-gogovapps** [comment]
@cyanglaz sanekyy rebased this for us again, can we get it merged?  Thanks!

**Comment by sanekyy** [comment]
@cyanglaz ping one more try

**Comment by jl-gogovapps** [comment]
@cyanglaz any word on if this can be merged?  Thanks!

**Comment by mattcoker** [comment]
Can someone please get this merged in? We're coming up on 16 months of this now.

**Comment by BabC** [comment]
@sanekyy @cyanglaz, any idea when this feature will be available ? =)

**Comment by cyanglaz** [comment]
@BabC Although this is important, this is currently not one of the features that we are actively working on. :) 

The PR doesn't link to any Github issues,  so it is also hard for us to triage and prioritize the PR.

**Comment by jl-gogovapps** [comment]
@cyanglaz I'm a little confused based on your comment previously:

> @sanekyy Sorry looks like this PR was ready to merge and it fell off my radar. Do you mind rebase the PR and ping me? I'll merge it.

**Review by cyanglaz — APPROVED** [review]
Left some comments, please fix those and I can land it. Sorry about the confused, I was confused this PR with a different PR. This PR is almost ready to go.

File: packages/google_maps_flutter/google_maps_flutter/pubspec.yaml [hunk 8]
```diff
@@ -1,13 +1,20 @@
...
+version: 0.5.31
 
 dependencies:
   flutter:
     sdk: flutter
   flutter_plugin_android_lifecycle: ^1.0.0
-  google_maps_flutter_platform_interface: ^1.0.4
+  google_maps_flutter_platform_interface: #^1.0.5
+    # workaround for success build checks in PR
+    git:
```
[the remark this answers is not in the document]
Comment (sanekyy) [reply packages/google_maps_flutter/google_maps_flutter/pubspec.yaml]: Removed. But now build checks will failed.

File: packages/google_maps_flutter/google_maps_flutter/lib/google_maps_flutter.dart [hunk 9]
```diff
@@ -46,4 +45,5 @@ export 'package:google_maps_flutter_platform_interface/google_maps_flutter_platf
         ScreenCoordinate;
 
 part 'src/controller.dart';
+
```
[the remark this answers is not in the document]
Comment (sanekyy) [reply packages/google_maps_flutter/google_maps_flutter/lib/google_maps_flutter.dart]: fixed

File: packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,7 @@
+## 0.5.31
```
[the remark this answers is not in the document]
Comment (sanekyy) [reply packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md]: done

**Comment by sanekyy** [comment]
@cyanglaz ping.

File: packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md [hunk 11]
```diff
@@ -1,3 +1,7 @@
+## 1.0.3
```
Comment (cyanglaz) [inline packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md:1]: should be 1.1.0

Comment (sanekyy) [reply packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md]: fixed

**Review by cyanglaz — COMMENTED** [review]
Thanks for the quick response. Sorry for the back and forth, but let's split the platform_interface change into a separate PR so we can publish the platform_interface first.

**Comment by sanekyy** [comment]
@cyanglaz done

**Comment by jl-gogovapps** [comment]
@sanekyy thanks again!

**Comment by fnicastri** [comment]
@cyanglaz 
Any news?

**Comment by jl-gogovapps** [comment]
has anyone else seen this plugin crash without a stack trace recently?  I'm starting to try to diagnose a reproducible crash (not necessarily based on this branch per se)

**Comment by fnicastri** [comment]
> has anyone else seen this plugin crash without a stack trace recently? I'm starting to try to diagnose a reproducible crash (not necessarily based on this branch per se)

We are using **this branch** in a production ready app (still in closed testing but with many users)
and 'till now @sanekyy 's branch is stable.

**Comment by jl-gogovapps** [comment]
Yea I've been using this branch since the '90s, definitely been stable and without issues. 😂 

My issues began yesterday, out of the blue.  I can reproduce the crash 100% of the time on iOS but can't get any error logs.

**Comment by fnicastri** [comment]
> Yea I've been using this branch since the '90s, definitely been stable and without issues. 😂
> 
> My issues began yesterday, out of the blue. I can reproduce the crash 100% of the time on iOS but can't get any error logs.

lol

can you tell us how to reproduce this ghost crash?

**Comment by jl-gogovapps** [comment]
> can you tell us how to reproduce this ghost crash?

My apologies, please disregard - I've narrowed the issue to between keyboard and chair 😄

**Comment by duzenko** [comment]
How do I use this in my app since the interface split?
If I checkout the interface branch, I get
```
error: The named parameter 'holes' isn't defined. (undefined_named_parameter at [pavement] lib\map\bidMeasure.dart:93)
```
If I checkout the holes branch, I get 
```
Because every version of google_maps_flutter from path depends on google_maps_flutter_platform_interface ^1.0.5 which doesn't match any versions, google_maps_flutter from path is forbidden.
So, because pavement depends on google_maps_flutter from path, version solving failed.
pub get failed (1; So, because pavement depends on google_maps_flutter from path, version solving failed.)```

**Comment by fnicastri** [comment]
> How do I use this in my app since the interface split?
> If I checkout the interface branch, I get
> 
> ```
> error: The named parameter 'holes' isn't defined. (undefined_named_parameter at [pavement] lib\map\bidMeasure.dart:93)
> ```
> 
> If I checkout the holes branch, I get
> 
> ```
> Because every version of google_maps_flutter from path depends on google_maps_flutter_platform_interface ^1.0.5 which doesn't match any versions, google_maps_flutter from path is forbidden.
> So, because pavement depends on google_maps_flutter from path, version solving failed.
> pub get failed (1; So, because pavement depends on google_maps_flutter from path, version solving failed.)```
> ```
> 
> My pubspec is
> 
> ```
>   google_maps_flutter:
>     path: ../flutterplugins/packages/google_maps_flutter/google_maps_flutter
> ```

You have to override the dep.

**Comment by jl-gogovapps** [comment]
Looks like cyanglaz approved this, are we almost able to get this PR merged in?

**Comment by cyanglaz** [comment]
@sanekyy https://github.com/flutter/plugins/pull/3135 was landed, do you mind update this PR according as well as resolving the conflicts? I will try to help you land this as soon as possible

**Pull request merged**